crates/noirc_evaluator/src/ssa/acir_gen/acir_mem.rs

```diff
@@ -25,21 +25,35 @@ use super::{
 /// Represent a memory operation on the ArrayHeap, at the specified index
 /// Operation is one for a store and 0 for a load
 #[derive(Clone, Debug)]
-struct MemOp {
+pub(crate) struct MemOp {
     operation: Expression,
     value: Expression,
     index: Expression,
 }
 
 type MemAddress = u32;
 
+enum ArrayType {
+    Init(HashSet<MemAddress>, u32),
+    WriteOnly,
+    ReadOnly(Option<usize>),
+    ReadWrite(Option<usize>),
```

Comment:
Can you add comments on each variant explaining what each field is for?

Reply:
Done

Reply:
Can you explain what the fields are in the comments as well? E.g. what does the `Option<usize>` represent? And each field in Init?

Reply:
Sure, I added more details.